apps/events/views.py

```diff
@@ -0,0 +1,123 @@
+from django.http import HttpResponseRedirect
+from django.shortcuts import get_object_or_404, render
+from django.urls import reverse
+
+from apps.events.forms import EventActionForm, StaticTriggerForm, TimeoutTriggerForm
+from apps.events.models import EventActionType, StaticTrigger, StaticTriggerType, TimeoutTrigger
+from apps.utils.time import seconds_to_human
+
+
+def create_timeout_event_view(request, team_slug: str, experiment_id: str):
+    return _create_event_view(TimeoutTriggerForm, request, team_slug, experiment_id)
+
+
+def create_static_event_view(request, team_slug: str, experiment_id: str):
+    return _create_event_view(StaticTriggerForm, request, team_slug, experiment_id)
+
+
+def _create_event_view(trigger_form_class, request, team_slug: str, experiment_id: str):
+    if request.method == "POST":
+        action_form = EventActionForm(request.POST)
+        if action_form.is_valid():
+            saved_action = action_form.save(experiment_id=experiment_id)
+            trigger_form = trigger_form_class(request.POST)
+            if trigger_form.is_valid():
```

Comment:
What happens if the trigger form isn't valid but the action form is? Can the user fix the errors and re-save or will that create 2 actions?

Reply:
Great point, will update